### ISSUE TITLE
Incompatibility with Dialogue UI

### ISSUE THREAD

**Issue by Fernando-Marquardt** [RestedXP/RXPGuides#271]
When running RestedXP with the Dialogue UI addon (https://www.curseforge.com/wow/addons/dialogueui), the addon struggles to track turn in steps.

Even after turning in quests, it is not detected by rested requiring to check the step manually. Sometimes it will go forward after accepting a new quest (not always) and sometimes after rotating the camera or hovering a NPC with the mouse.

This is an example where 'Tools for Steelgrill' is already completed and even after talking with another NPC to accept 'Ammo for Rumbleshot', rested still does not detect the quest completion.
![image](https://github.com/user-attachments/assets/fb47a680-d9b4-40cc-af47-ec3e3987ca7b)

**Comment by Starlynk1** [comment]
Yup, seeing this happen as well. I've turned off the auto turnin option in DialogueUI and still does not track turn ins.

**Comment by Starlynk1** [comment]
Found something interesting in regards to RXP and possibly Dialogue UI. If the turn in was not showing complete and I sat there a few seconds, RXP did eventually update that it was turned in. I am guessing that RXP has a type of 'fail safe' that checks periodically the state of the current step.

Continueing to monitor that and see if it happens every time it misses a turn in and then corrects itself.

**Comment by Fernando-Marquardt** [comment]
> Found something interesting in regards to RXP and possibly Dialogue UI. If the turn in was not showing complete and I sat there a few seconds, RXP did eventually update that it was turned in. I am guessing that RXP has a type of 'fail safe' that checks periodically the state of the current step.
> 
> Continueing to monitor that and see if it happens every time it misses a turn in and then corrects itself.

Yes, this does happen sometimes. But there are some quests that never tick complete, no matter how much time you wait or what you do (sat there for half an hour and nothing happened, ran from Kharanos to Iron Forge and nothing happened either), other than ticking it manually.

Sometimes it does not complete or accept the quest as well, it just selects the quest but it never moves on. What I find it weird, the same quests always fail and the ones working, always work. So not sure if a delay on the addon could be the issue.

I have tried other guide addons, but RestedXP seems to be the only one having issues tracking quests with Dialogue UI active.

Maybe I will take a look at the code myself and try to find the issue. It must be something with Rested trying to be faster than the dialog before checking the status.

**Comment by Starlynk1** [comment]
Yeah, waited on one and it never completed. Hopefully the devs can hop on this soon and find a fix.
